### ISSUE TITLE
About "Please recommend additional papers, blog posts, and implementations."

### ISSUE THREAD

**Issue by mth** [tekknolagi/tekknolagi.github.com#77]
Using the issues as a comment form for https://bernsteinbear.com/blog/row-poly/ .

I think I've implemented something similar in Yeti. I also documented the algorithm that I wrote (I didn't know then that it is basically a form of row polymorphism).

https://dot.planet.ee/yeti/reference.html#structure-and-variant-types
https://dot.planet.ee/yeti/reference.html#finding-free-type-variables

The "Restricting the free type variables" part is important there for the algorithm to work with mutable record fields.

The implementation itself in Java is a bit hard to follow, but can link it nevertheless.

https://github.com/mth/yeti/blob/master/c/YetiType.java#L535
https://github.com/mth/yeti/blob/master/c/YetiType.java#L999

**Comment by tekknolagi** [comment]
Sorry, I will get to this at some point
